server/drivers/hd44780.c

```diff
@@ -466,23 +666,74 @@ HD44780_init(Driver *drvthis)
 void
 common_init(PrivateData *p, unsigned char if_bit)
 {
-	if (p->ext_mode) {
+	/* Set initial brightness according to Brightness setting.
+	 * This assumes that initially backlight is on (or if not used at all) */
+	int init_brightness = p->brightness;
+
+	unsigned char cmd_funcset =  FUNCSET | if_bit | TWOLINE | SMALLCHAR;
+	if (has_extended_mode(p)) {
 		/* Set up extended mode */
-		p->hd44780_functions->senddata(p, 0, RS_INSTR, FUNCSET | if_bit | TWOLINE | SMALLCHAR | EXTREG);
+		p->hd44780_functions->senddata(p, 0, RS_INSTR, cmd_funcset | EXTREG);
 		p->hd44780_functions->uPause(p, 40);
 		p->hd44780_functions->senddata(p, 0, RS_INSTR, EXTMODESET | FOURLINE);
 		p->hd44780_functions->uPause(p, 40);
 	}
-	p->hd44780_functions->senddata(p, 0, RS_INSTR, FUNCSET | if_bit | TWOLINE | SMALLCHAR | p->font_bank);
+
+	if (p->model == HD44780_MODEL_PT6314_VFD) {
+		cmd_funcset &= ~PT6314_BRIGHT_MASK;
+		p->func_set_mode = cmd_funcset;
```

Comment:
This line seems redundant to the same line below ...

Reply:
You are right.